src/d3a/models/market/two_sided.py

```diff
@@ -264,9 +277,70 @@ def match_recommendation(self, recommended_list):
                     recommended_list, index+1, trade, bid_trade
                 )
 
+    @staticmethod
+    def validate_requirements_satisfied(obj1: Union[Offer, Bid], obj2: Union[Offer, Bid]) -> None:
```

Comment:
This logic seems overly complex to be in a function, and I think it would be a very good candidate for a separate class (e.g. `RequirementsValidator` or `RequirementsComplianceValidator`, or any better name).
Actually, the validation of each individual requirement could also deserve its own class instead of having so many conditionals. When requirements will increase this function will become almost impossible to expand and test.

We're using generic objects `obj1` and `obj2`, and I have the feeling that this is complicating the internal logic. Could you explain why this is implemented this way? Isn't it possible to distinguish between offers/bids? This would probably remove the need for some conditionals and `hasattr` checks.

Reply:
Thanks for putting my doubts into proper words! I second that.

Reply:
I agree, I thought about this but then ignored it while implementing and focused on making things work, anw I'll create something that doesn't break the open for extension closed for modification principle